test_conformance/extensions/cl_khr_command_buffer/negative_command_buffer_get_info.cpp

```diff
@@ -0,0 +1,312 @@
+            &number_of_queues, nullptr);
+        test_error(error, "Unable to query CL_COMMAND_BUFFER_NUM_QUEUES_KHR");
+
+
+        size_t ret_value_size = 0;
+        error = clGetCommandBufferInfoKHR(
+            command_buffer, CL_COMMAND_BUFFER_PROPERTIES_ARRAY_KHR, 0, nullptr,
+            &ret_value_size);
+        test_error(error,
+                   "Unable to query CL_COMMAND_BUFFER_PROPERTIES_ARRAY_KHR");
+
+        number_of_properties =
+            ret_value_size / sizeof(cl_command_buffer_properties_khr);
+
+        return CL_SUCCESS;
+    }
+
+    cl_int RunParamNameTest()
+    {
+        cl_int error = clGetCommandBufferInfoKHR(
+            command_buffer, CL_COMMAND_BUFFER_FLAGS_KHR, sizeof(cl_uint),
+            nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_VALUE,
+                               "clGetCommandBufferInfoKHR should return "
+                               "CL_INVALID_VALUE",
+                               TEST_FAIL);
+
+        cl_uint ret_val = 0;
+
+        error = clGetCommandBufferInfoKHR(
+            command_buffer, CL_COMMAND_BUFFER_FLAGS_KHR, 0, &ret_val, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_VALUE,
+                               "clGetCommandBufferInfoKHR should return "
+                               "CL_INVALID_VALUE",
+                               TEST_FAIL);
+
+        error = clGetCommandBufferInfoKHR(command_buffer,
+                                          CL_COMMAND_BUFFER_NUM_QUEUES_KHR, 0,
+                                          &ret_val, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_VALUE,
+                               "clGetCommandBufferInfoKHR should return "
+                               "CL_INVALID_VALUE",
+                               TEST_FAIL);
+
+        return TEST_PASS;
+    }
+
+    cl_int RunQueuesInfoTest()
+    {
+        cl_uint num_queues = 0;
+
+        cl_int error = clGetCommandBufferInfoKHR(
+            command_buffer, CL_COMMAND_BUFFER_NUM_QUEUES_KHR, 0, &num_queues,
+            nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_VALUE,
+                               "clGetCommandBufferInfoKHR should return "
+                               "CL_INVALID_VALUE",
+                               TEST_FAIL);
+
+        cl_command_queue queues[number_of_queues];
+        error = clGetCommandBufferInfoKHR(
+            command_buffer, CL_COMMAND_BUFFER_QUEUES_KHR, 0, queues, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_VALUE,
+                               "clGetCommandBufferInfoKHR should return "
+                               "CL_INVALID_VALUE",
+                               TEST_FAIL);
+
+        return TEST_PASS;
+    }
+
+    cl_int RunRefCountInfoTest()
+    {
+        cl_uint ref_count = 0;
+
+        cl_int error = clGetCommandBufferInfoKHR(
+            command_buffer, CL_COMMAND_BUFFER_REFERENCE_COUNT_KHR, 0,
```

Comment:
Can we use a non-zero `param_value_size` in these tests too? It's possible that an implementation could have a check against zero but not against being less that the return type in general.

For example, here we could use `sizeof(ref_count)-1` as another `param_value_size` input too.